core/src/test/java/org/apache/iceberg/rest/TestHTTPHeaders.java

```diff
@@ -95,6 +100,9 @@ void addIfAbsentHTTPHeader() {
                 "header1", List.of("value1a", "value1b"),
                 "header2", List.of("value2"),
                 "header3", List.of("value3")));
+
+    assertThatThrownBy(() -> headers.addIfAbsent((HTTPHeaders) null))
+        .isInstanceOf(NullPointerException.class);
```

Comment:
checks like this should always have a `.hasMessage` check to make sure we get the right error msg back (unfortunately we can't easily enforce such a rule via checkstyle)

Reply:
The message is generated by Immutables and will be rather cryptic. But OK.